### PR TITLE
Add `LightClientHeader` wrapper

### PR DESCRIPTION
**Pull request by etan-status**
In Altair, light client sync protocol exchanges `BeaconBlockHeader` structures for tracking current progress. Wrapping `BeaconBlockHeader` inside a `LightClientHeader` allows future extensions of this header, e.g., to also track `ExecutionPayloadHeader`.

Note: This changes the JSON REST format by adding a `beacon` nesting. For SSZ, the serialization format stays same (but overall root changes).

### CODE REVIEW

**Review by hwwhww — COMMENTED** [review]
Although the fact that it modifying the Altair spec is kinda cheating, I think it is cleaner for consensus specs. So I personally prefer this PR over #3151 solution.

/cc @dapplion, @wemeetagain for third opinions. The discussion was here: https://github.com/ethereum/consensus-specs/pull/3151#pullrequestreview-1236859997

**Comment by etan-status** [comment]
The SSZ format stays the same, which is key, as otherwise we would break compatibility of gossip, potentially disconnecting nodes. As for JSON, clients will have to learn how to deal with the extra `beacon` nesting from Capella onwards anyway, so their work to update would be necessary eventually (or they could just use SSZ to begin with).

Note that no release for beacon-APIs was cut yet that formally locks in the JSON endpoints.

File: specs/altair/light-client/sync-protocol.md [hunk 1]
```diff
@@ -73,13 +74,21 @@ Additional documents describe how the light client sync protocol can be used:
 
 ## Containers
 
+### `LightClientHeader`
+
+```python
+class LightClientHeader(Container):
+    # Beacon block header
+    beacon: BeaconBlockHeader
+```
```
[the remark this answers is not in the document]
Comment (etan-status) [reply specs/altair/light-client/sync-protocol.md:84]: Added comments, and also the stub for `is_valid_light_client_header`

**Comment by etan-status** [comment]
Actually, this one would kill the entirety of #3150 until `BeaconBlockHeader` extends (in which case, `get_lc_beacon_root` would have to be introduced once more). So, another reason to take this one for simplicity.

**Review by dapplion — APPROVED** [review]
- I think it's worth it to make a breaking change on altair spec now that adoption is low to prevent more complexity via legacy code. I'm in favor of https://github.com/ethereum/consensus-specs/pull/3151 (see my comment for rationale) so we should commit in that direction now

**Comment by hwwhww** [comment]
@dapplion 
just to confirm: you want "Add ExecutionPayloadHeader to LC data" feature, and per the compatibility approaches, you like this PR's  (add `LightClientHeader` wrapper)'s approach rather than the #3150 approach, correct?

**Pull request merged**